core/server/master/src/main/java/alluxio/master/throttle/ServerIndicator.java

```diff
@@ -0,0 +1,270 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The server indicators.
+ */
+public class ServerIndicator {
+  private static final Logger LOG = LoggerFactory.getLogger(ServerIndicator.class);
+  private Long mHeapMax;
+  private Long mHeapUsed;
+  private long mDirectMemUsed;
+  private Double mCpuLoad;
+
+  private long mTotalJVMPauseTimeMS;
+  private long mPITTotalJVMPauseTimeMS;
+  private long mRpcQueueSize;
+
+  private long mPitTimeMS;
+
+  /**
+   * @param directMemUsed  the allocated direct memory
+   * @param heapMax the max heap
+   * @param heapUsed the used heap
+   * @param cpuLoad the cpu load
+   * @param totalJVMPauseTimeMS the total JVM pause time since previous PIT
+   * @param pitTotalJVMPauseTimeMS the pit total JVM pause time
+   * @param rpcQueueSize the pit rpc queue size
+   * @param snapshotTimeMS the pit time
+   */
+  public ServerIndicator(long directMemUsed, long heapMax, long heapUsed, double cpuLoad,
+      long totalJVMPauseTimeMS, long pitTotalJVMPauseTimeMS, long rpcQueueSize,
+      long snapshotTimeMS) {
+    mDirectMemUsed = directMemUsed;
+    mHeapMax = heapMax;
+    mHeapUsed = heapUsed;
+    mCpuLoad = cpuLoad;
+    mTotalJVMPauseTimeMS = totalJVMPauseTimeMS;
+    mPITTotalJVMPauseTimeMS = pitTotalJVMPauseTimeMS;
+    mRpcQueueSize = rpcQueueSize;
+    mPitTimeMS = snapshotTimeMS;
+  }
+
+  /**
+   * @param serverIndicator the server indicator
+   */
+  public ServerIndicator(ServerIndicator serverIndicator) {
+    this(serverIndicator, 1);
+  }
+
+  /**
+   * The scaled server indicator according to the multiple.
+   *
+   * @param serverIndicator the base server indicator
+   * @param multiple the times
+   */
+  public ServerIndicator(ServerIndicator serverIndicator, int multiple) {
+    Preconditions.checkNotNull(serverIndicator, "serverIndicator");
+    mDirectMemUsed = serverIndicator.mDirectMemUsed * multiple;
+    mHeapMax = serverIndicator.mHeapMax;
+    mHeapUsed = serverIndicator.mHeapUsed * multiple;
+    mCpuLoad = serverIndicator.mCpuLoad * multiple;
+    mTotalJVMPauseTimeMS = serverIndicator.mTotalJVMPauseTimeMS * multiple;
+    mPITTotalJVMPauseTimeMS = serverIndicator.mPITTotalJVMPauseTimeMS;
+    mRpcQueueSize = serverIndicator.mRpcQueueSize * multiple;
+    mPitTimeMS = serverIndicator.mPitTimeMS;
+  }
+
+  /**
+   * @return current total JVM pause time
+   */
+  public static long getSystemTotalJVMPauseTime() {
+    if (Configuration.getBoolean(PropertyKey.MASTER_JVM_MONITOR_ENABLED)) {
+      return (long) (MetricsSystem.METRIC_REGISTRY
+          .gauge(MetricsMonitorUtils.ServerGaugeName.TOTAL_EXTRA_TIME, null).getValue());
+    }
+    return 0;
+  }
+
+  /**
+   * creates server indicator based on current status.
+   *
+   * @param prevTotalJVMPauseTime the previous total JVM pause time
+   * @return the server indicator
+   */
+  public static ServerIndicator createFromMetrics(long prevTotalJVMPauseTime) {
+    long pitTotalJVMPauseTime = 0;
```

Comment:
I think its better code style to put these closer to where they are set.

Reply:
Modified a little